### ISSUE TITLE
Support GHC 9.0 in generated projects

### ISSUE THREAD

**Issue by vrom911** [kowainik/summoner#539]

**Comment by ArturGajowy** [comment]
Is there anything that prevents resolving the available GHC versions dynamically? Say, from ghcup, stack lts, or even adding the one available on PATH? I've seen there's support for 8.10 implemented, but I didn't get it from a fresh `cabal install` (v2.1 is not released yet?).

```
Additionally you can specify versions of GHC to test with (space-separated): 
  ⓘ Supported by 'summoner' GHCs: 7.10.3 8.0.2 8.2.2 8.4.4 8.6.5 8.8.3
 ** SNIP **

Job's done
✔ ~/projects 
14:50 $ ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.10.4
✔ ~/projects 
14:50 $ summon --version
Summoner v2.0.1.1
 ➤ Git revision: UNKNOWN
 ➤ Commit date:  UNKNOWN
```

**Comment by chshersh** [comment]
@ArturGajowy I [wrote a comment](https://github.com/kowainik/summoner/issues/535#issuecomment-791944112) previously, providing an explanation of why config-based approach will result in bad UX. The same is true for the dynamic discovery of GHC.

Yes, 2.1 is not on Hackage yet. If you want to use the latest Summoner version, you can [install it from sources](https://github.com/kowainik/summoner#tui-from-source).
